brainiak/funcalign/sssrm.py

```diff
@@ -0,0 +1,809 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Semi-Supervised Shared Response Model (SS-SRM)
+
+The implementations are based on the following publications:
+
+.. [Turek2016] "A Semi-Supervised Method for Multi-Subject fMRI Functional
+   Alignment",
+   J. Turek, T. Willke, P.-H. Chen, P. Ramadge
+   under review, 2016.
+"""
+
+# Authors: Javier Turek (Intel Labs), 2016
+
+import logging
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin, ClassifierMixin
+from sklearn.utils import assert_all_finite
+from sklearn.utils.validation import NotFittedError
+from sklearn.utils.multiclass import unique_labels
+import theano
+import theano.tensor as T
+import theano.compile.sharedvalue as S
+from pymanopt.manifolds import Euclidean
+from pymanopt.manifolds import Product
+from pymanopt.solvers import ConjugateGradient
+from pymanopt import Problem
+from pymanopt.manifolds import Stiefel
+import gc
+
+from brainiak.utils import utils
+from brainiak.funcalign import srm
+
+__all__ = [
+    "SSSRM"
+]
+
+logger = logging.getLogger(__name__)
+
+
+class SSSRM(BaseEstimator, ClassifierMixin, TransformerMixin):
+    """Semi-Supervised Shared Response Model (SS-SRM)
+
+    Given multi-subject data, factorize it as a shared response S among all
+    subjects and an orthogonal transform W per subject, using also labeled
+    data to train a Multinomial Logistic Regression (MLR) classifier (with
+    l2 regularization) in a semi-supervised manner:
+
+    .. math:: X_i \\approx W_i S ,~for~all~i=1\dots N
```

Comment:
Is there a reason we write the math like this instead of the objective function? It's a bit unclear how do the parameters affect the model. In other words, there's no gamma and "cost" in the math.